packages/api/src/types.ts

```diff
@@ -132,6 +132,13 @@ export interface ApiOptions {
    * uses types not available in the base Substrate runtime.
    */
   types?: RegistryTypes;
+  /**
+   * @description prebundles is a map of 'genesis hash and runtime spec version' as key to metadata's hex string
+   * if genesis hash and runtime spec version matches, then use metadata, else fetch it from chain
+   */
+  prebundles?: {
```

Comment:
Not sold on the naming of `prebundles` at all. Doesn't tell me what it is. `source.metadata[${genesisHash}-${specVersion}]` works as well

Reply:
Thanks for your comment.. I believe I can't simply put metadata(adding Partial to ApiBase<any, any>) in source as doing that, I would also need other things like _isReady, genesisHash, runtimeVersion. 
For now I renamed prebundles to metadata.